packages/expo-modules-core/common/cpp/fabric/ExpoViewShadowNode.cpp

```diff
@@ -31,6 +31,12 @@ void ExpoViewShadowNode::initialize() noexcept {
   } else {
     traits_.unset(react::ShadowNodeTraits::Trait::ChildrenFormStackingContext);
   }
+  // This is needed for display: contents, so that native view is still rendered
+  // https://github.com/facebook/react-native/blob/b02251e7f5c147296fab93c1ae613d27400cec92/packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp#L399
+  // Without it the native view is flattened and not added to native hierarchy
+  if (viewProps.collapsable == false) {
```

Comment:
```suggestion
  // TODO(Ubax,20250830): This change should be upstreamed and removed when https://github.com/facebook/react-native/pull/N were landed
  if (viewProps.collapsable == false) {
```

i feel the [fix](https://github.com/facebook/react-native/commit/bb82e16a278693f27d26769e75d294881e176c5a) should be upstreamed. (correct me if my understanding is not right)
if it should be upstreamed and we need this workaround first, then we should create the upstream pr and add a todo note to clean it up later

Reply:
i think we should do [this](https://github.com/expo/expo/pull/39262) for now instead of `collapsable` as there could be cases where it can have unexpected behaviour. (e.g. overflow hidden + display contents makes the view invisible).